state/txn_test.go

```diff
@@ -1,128 +1,52 @@
 package state
 
 import (
-	"bytes"
-	"crypto/rand"
 	"fmt"
 	"math/big"
 	"testing"
 
-	"github.com/0xPolygon/polygon-edge/helper/hex"
 	"github.com/0xPolygon/polygon-edge/types"
 	"github.com/stretchr/testify/assert"
-	"github.com/umbracle/fastrlp"
-	"golang.org/x/crypto/sha3"
 )
 
-type mockState struct {
-	snapshots map[types.Hash]Snapshot
+type mockSnapshot struct {
+	state map[types.Address]*PreState
 }
 
-func (m *mockState) NewSnapshotAt(root types.Hash) (Snapshot, error) {
-	t, ok := m.snapshots[root]
+func (m *mockSnapshot) GetStorage(addr types.Address, root types.Hash, key types.Hash) types.Hash {
+	raw, ok := m.state[addr]
 	if !ok {
-		return nil, fmt.Errorf("not found")
+		return types.Hash{}
 	}
-
-	return t, nil
-}
-
-func (m *mockState) NewSnapshot() Snapshot {
-	return &mockSnapshot{data: map[string][]byte{}}
-}
-
-func (m *mockState) GetCode(hash types.Hash) ([]byte, bool) {
-	panic("Not implemented in tests")
-}
-
-type mockSnapshot struct {
-	data map[string][]byte
-}
-
-func (m *mockSnapshot) Get(k []byte) ([]byte, bool) {
-	v, ok := m.data[hex.EncodeToHex(k)]
-
-	return v, ok
-}
-
-func (m *mockSnapshot) Commit(objs []*Object) (Snapshot, []byte) {
-	panic("Not implemented in tests")
+	res, ok := raw.State[key]
+	if !ok {
+		return types.Hash{}
+	}
+	return res
 }
 
-func newStateWithPreState(preState map[types.Address]*PreState) (*mockState, *mockSnapshot) {
-	state := &mockState{
-		snapshots: map[types.Hash]Snapshot{},
-	}
-	snapshot := &mockSnapshot{
-		data: map[string][]byte{},
+func (m *mockSnapshot) GetAccount(addr types.Address) (*Account, error) {
+	raw, ok := m.state[addr]
+	if !ok {
+		return nil, fmt.Errorf("account not found")
 	}
-
-	ar := &fastrlp.Arena{}
-
-	for addr, p := range preState {
-		account, snap := buildMockPreState(p)
-		if snap != nil {
-			state.snapshots[account.Root] = snap
-		}
-
-		v := account.MarshalWith(ar)
-		accountRlp := v.MarshalTo(nil)
-		/*
-			accountRlp, err := rlp.EncodeToBytes(account)
-			if err != nil {
-				panic(err)
-			}
-		*/
-		snapshot.data[hex.EncodeToHex(hashit(addr.Bytes()))] = accountRlp
+	acct := &Account{
+		Balance: new(big.Int).SetUint64(raw.Balance),
+		Nonce:   raw.Nonce,
 	}
-
-	return state, snapshot
+	return acct, nil
 }
 
-func newTestTxn(p map[types.Address]*PreState) *Txn {
-	return newTxn(newStateWithPreState(p))
+func (m *mockSnapshot) GetCode(hash types.Hash) ([]byte, bool) {
```

Comment:
Do we need second output parameter instead of returning 'nil' or we might have 'nil' values as well?